### DISCUSSION TITLE
How do you show the dimensions for a text label or caption?

### DISCUSSION THREAD

**Discussion by John-Whitehead** [ImageMagick/ImageMagick#6414] [Digital Image Processing] [answered]
What I want to do:

When writing text on top of an image, I would like to know the dimensions of the block of text so I can calculate the brightness of the part of the image under that text to help decide to use a light or dark color font.

What I have tried:

Using Imagemagick, I have the steps all worked out, EXCEPT for getting the dimensions of the block of text. I have a long string and want it broken up (new lines) around the 300 pixel mark.

This seems to be a bit above my pay grade, so I asked ChatGPT. It gave me this command:

convert -size 300x -font Arial -pointsize 20 -gravity center -fill black -caption "This is a long text label that needs to be split into multiple lines" -density 96 -trim +repage -bordercolor white -border 10x10 -format "%@" info:

Unfortunately, it does not work and gives me the error:

convert: no images defined `info:' @ error/convert.c/ConvertImageCommand/3342.

I have tried a bunch of changes, but the only time get any output, it is something useless like 1x1+1+1

So, my question is: Using Imagemagick, how can I get the size of a caption or text label?

**Comment by fmw42** [comment, marked as answer]
Change -caption to caption: and remove the space to the text.  @ is only appropriate if you do not use -trim

```
convert -size 300x -font Arial -pointsize 20 -gravity center -fill black caption:"This is a long text label that needs to be split into multiple lines" -density 96 -trim +repage -bordercolor white -border 10x10 -format "%wx%h" info:
277x83
```
